kubernetes/config-maps/nginx-storage/default.conf.j2

```diff
@@ -8,12 +10,31 @@ server {
     client_max_body_size     {{ max_upload_size }};
 
 
+    set $gatekeeper_api_backend http://gatekeeper.default.svc.cluster.local;
+
     location /download/ {
-        limit_except GET { deny all; }
+        limit_except        GET { deny all; }
+        auth_request        /download-auth/;
+        auth_request_set    $auth_status $upstream_status;
+        if ($request_uri ~* "/download/(.*)") {
```

Comment:
1) Why case-insensitive regex match (`~*`) and and not a normal one (`~`)?
2) What will match be if `request_uri` looks like this: `/download/a/b/c/download/file.jpg`?

Reply:
1. According to this post https://serverfault.com/questions/801356/what-is-the-difference-between-nginx-and-regexes, version with `~*` is matching as a case-insensitive regular expression.
2.  So if request look like this `download/a/b/c/download/file.jpg`, it change properly and go to gatekeeper in this form of request `/gatekeeper/download-auth/a/b/c/download/file.jpg`

Reply:
> According to this post https://serverfault.com/questions/801356/what-is-the-difference-between-nginx-and-regexes, version with ~* is matching as a case-insensitive regular expression.

I'm asking why do you need a case-insensitive match here.